core/treeclass.py

```diff
@@ -1562,8 +1562,8 @@ def data_socket(bsocket):
         else:
             class_name = constants.CLASS_NAMES[socket_type]
 
-        exec(f"from geonodes import {class_name}", locals(), globals())
-        return eval(f"{class_name}(bsocket)", locals(), globals())
+        mod = __import__('', globals(), locals(), [class_name], 2)
```

Comment:
This probably could use `import . as core` as well and then `getattr(core, class_name)`?